app/controllers/api/v1/npq_applications_controller.rb

```diff
@@ -59,6 +61,13 @@ def filter
         params[:filter] ||= {}
       end
 
+      def validate_filter_param
+        unless filter.as_json.is_a?(Hash)
+          error_factory = Api::ParamErrorFactory.new(params: ["Filter must be a hash"], error: "Bad parameter")
+          render json: { errors: error_factory.call }, status: :bad_request
+        end
+      end
```

Comment:
Should we have this check in `ApiFilter`? Perhaps when we try to access the filter we could check its format, and raise an error that will get caught by api controller error handling?

That way we can add this check to every place that is using `filter` url parameter in one go.

Reply:
Yeah, already on it. That's why I waited for your stuff to go in first. I've tweaked and rebased